src/query/phrase_prefix_query/phrase_prefix_weight.rs

```diff
@@ -0,0 +1,257 @@
+use super::{prefix_end, PhrasePrefixScorer};
+use crate::core::SegmentReader;
+use crate::fieldnorm::FieldNormReader;
+use crate::postings::SegmentPostings;
+use crate::query::bm25::Bm25Weight;
+use crate::query::explanation::does_not_match;
+use crate::query::{EmptyScorer, Explanation, Scorer, Weight};
+use crate::schema::{IndexRecordOption, Term};
+use crate::{DocId, DocSet, Score};
+
+pub struct PhrasePrefixWeight {
+    phrase_terms: Vec<(usize, Term)>,
+    prefix: (usize, Term),
+    similarity_weight_opt: Option<Bm25Weight>,
+    max_expansions: u32,
+}
+
+impl PhrasePrefixWeight {
+    /// Creates a new phrase weight.
+    /// If `similarity_weight_opt` is None, then scoring is disabled
+    pub fn new(
+        phrase_terms: Vec<(usize, Term)>,
+        prefix: (usize, Term),
+        similarity_weight_opt: Option<Bm25Weight>,
+        max_expansions: u32,
+    ) -> PhrasePrefixWeight {
+        PhrasePrefixWeight {
+            phrase_terms,
+            prefix,
+            similarity_weight_opt,
+            max_expansions,
+        }
+    }
+
+    fn fieldnorm_reader(&self, reader: &SegmentReader) -> crate::Result<FieldNormReader> {
+        let field = self.phrase_terms[0].1.field();
+        if self.similarity_weight_opt.is_some() {
+            if let Some(fieldnorm_reader) = reader.fieldnorms_readers().get_field(field)? {
+                return Ok(fieldnorm_reader);
+            }
+        }
+        Ok(FieldNormReader::constant(reader.max_doc(), 1))
+    }
+
+    pub(crate) fn phrase_scorer(
+        &self,
+        reader: &SegmentReader,
+        boost: Score,
+    ) -> crate::Result<Option<PhrasePrefixScorer<SegmentPostings>>> {
+        let similarity_weight_opt = self
+            .similarity_weight_opt
+            .as_ref()
+            .map(|similarity_weight| similarity_weight.boost_by(boost));
+        let fieldnorm_reader = self.fieldnorm_reader(reader)?;
+        let mut term_postings_list = Vec::new();
+        if reader.has_deletes() {
+            for &(offset, ref term) in &self.phrase_terms {
+                if let Some(postings) = reader
+                    .inverted_index(term.field())?
+                    .read_postings(term, IndexRecordOption::WithFreqsAndPositions)?
+                {
+                    term_postings_list.push((offset, postings));
+                } else {
+                    return Ok(None);
+                }
+            }
+        } else {
+            for &(offset, ref term) in &self.phrase_terms {
+                if let Some(postings) = reader
+                    .inverted_index(term.field())?
+                    .read_postings_no_deletes(term, IndexRecordOption::WithFreqsAndPositions)?
```

Comment:
I know you copied this from phrase_weight but the code is meaningless.. As far as I can tell read_postings_no_deletes does the same as read_postings.

Can you open an issue to cleanup this mess `read_postings_no_deletes`?

Correctness is (probably) not at stake here... We do the check at the collector level anyway.
There is a non-trivial optimization decision to take here.

Do we do 

check_phrase(intersection(remove_deletes(postings)))
or
remove_deletes(check_phrase(intersection(postings)))


The earlier probably seems trivially faster, but I suspect it depends on the filtering power of deletes.
Anyway for the moment we can probably stick to 

remove_deletes(check_phrase(intersection(postings)))

for simplificity, and clean up the existing code.